activerecord/lib/active_record/errors.rb

```diff
@@ -7,10 +7,6 @@ module ActiveRecord
   class ActiveRecordError < StandardError
   end
 
-  # Raised when trying to use a feature in Active Record which requires Active Job but the gem is not present.
-  class ActiveJobRequiredError < ActiveRecordError
```

Comment:
If we want to remove this exception we need to deprecate first.

Reply:
I added a deprecation:

https://github.com/rails/rails/pull/45476/files#diff-805517f2e613a421493817fa15ecd42e583caf1b4f6f6972ec6e8103f41e04a2R12

...and a test:

https://github.com/rails/rails/pull/45476/files#diff-0045b3212d324439c6e30bef62094dcf2654863b9ccd93ef2367cec9f135e044R42-R49